### ISSUE TITLE
New way of manage player list

### ISSUE THREAD

**Issue by thejaviertc** [PazerOP/tf2_bot_detector#285]
**Basic explanation of the idea:**
Instead of having an official list and several unofficial lists, where there is no communication between lists, it would be better to create a single one, in which anyone who has found a bot / hacker, can add him to the list, but it would lead to some problems that it would have to be solved before.

**Problems and Solutions of the idea:**
The main problem is obvious, if anyone can add someone to the list, it would surely add innocent people. This is where the idea stands out.

You could create a system of "reports" in which when you add a bot / hacker to the list, it would be added to a list of suspects with the amount of "reports" from different people. If it reaches x amount, it would automatically be added to another list (or whatever) that will include all the bots / cheaters that really are, since too many people should be gathered to be able to add an innocent person there.

In the application you could use the category of suspects for users who have been "reported" for the first time, and the category of hacker for those who have already been "reported" exceeding x amount.

Thanks to this, all the users who are participating in this project would work together to create a single list that contains the vast majority of bots / hackers, and the efforts will not be divided into lists of other users, or creating your own.

**PS: Today I discovered this project and I loved it, it is the first time that I see something of such a level to try to end the bot problem in Team Fortress 2. You are doing a great job.**

**Comment by brachistosaurus** [comment]
What happens if a bot army mass-reports a single, innocent player?

**Comment by Kenajcrap** [comment]
The system you discribed (even with the "solution") would be *extremely* abusable to anyone with I'll intentions. It takes the same effort for a bit hoster to use one instance of the program to report a player than it does for him to use dozens.

"Just make so if the same instances keep report a single player, their reports are not counted"
Attackers can just fuzzy up what instances are used to report, and not use very instance every time

"Just make so the threshold for being marked as a cheater is higher"
The program is pretty light, Im sure someone will be able to run hundreds of instances

"Just authenticate (somehow) and prevent multiple instances (somehow)"
Normal players are suuuper trigger happy when it comes to an enemy cheaters, making the list unified would create A LOT of false positives, even if multiple reports are needed to mark someone as cheater. One of the main reasons local marks are not propagated in any way is to curb the impact of false positives that will inevitably happen because some players are dumb.

"Just do all of the above, and also create a smarter system to evaluate the 'worth' of each report based on factors like how frequently the player reports, how frequently the other player is reported, build a statistical model to measure how likely a player with those profile information items is to be cheating, etc, so false positives would be minimal"
Something like this would definitely be out of scope for this project.

Basically while this is a good idea the first time you think about it, it is not possible to prevent abuse effectively and creating something like this would be a very bad idea and very hard to manage. Basically: this is a job for valve.

**Comment by thejaviertc** [comment]
Ok, @Kenajcrap have just broken my idea in 5 minutes.

I didn't imagine almost all the things he said.

Anyways, that could be a bad idea, but I think they have to somehow join all list at once, but maybe it could be a good idea to make a "system" where moderators selected by PazerOP could see videos/screenshots of the reported bots, and if it's safe that is a bot/hacker, add him to the list. All the users would have the same list, and the users could contribute to make the list bigger.

**Comment by brachistosaurus** [comment]
> it could be a good idea to make a "system" where moderators selected by PazerOP could see videos/screenshots of the reported bots

There is the **Trusted list**, but it's "hidden" away from most players (in the Wiki). Probably to limit the impact of false positives. I don't know if it's still updated, and I don't know if there are checks and balances.

**Comment by ClusterConsultant** [comment]
This has been discussed at length in many many places. This project simply does not have the resources to even begin approaching this idea. The list format is open. If someone wants to create and moderate this new community idea they are welcome but this is far beyond the scope of what we can do here.
